File: src/main/java/com/cuupa/dms/storage/document/db/DocumentMapper.java
package com.cuupa.dms.storage.document.db;

import com.cuupa.dms.storage.tag.Tag;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class DocumentMapper {

    public static Document mapToEntity(com.cuupa.dms.storage.document.Document document) {
        Document
                documentEntity =
                new Document(document.getId(),
                             document.getName(),
                             document.getSender(),
                             document.getOwner(),
                             document.getCreateDate(),
                             new ArrayList<>());
        documentEntity.setImage(document.getImage());
        return documentEntity;
    }

    public static com.cuupa.dms.storage.document.Document mapToGuiObject(Document document, List<Tag> tags) {
        com.cuupa.dms.storage.document.Document
                uiDocument =
                new com.cuupa.dms.storage.document.Document(document.getId(),
                                                            document.getName(),
                                                            document.getSender(),
                                                            document.getOwner(),
                                                            document.getCreateDate(),
                                                            tags.stream()
                                                                .map(tag -> new Tag(tag.getId(),
                                                                                    tag.getName(),
                                                                                    tag.getOwner()))
                                                                .collect(Collectors.toList()));
        uiDocument.setImage(document.getImage());
        return uiDocument;
    }
}

File: src/main/java/com/cuupa/dms/storage/document/db/DocumentMapper.kt
package com.cuupa.dms.storage.document.db

import com.cuupa.dms.storage.document.Document
import com.cuupa.dms.storage.tag.Tag

object DocumentMapper {

    fun mapToEntity(document: Document): DBDocument {
        return DBDocument(document.filename,
                document.name,
                document.sender,
                document.owner,
                document.createDate,
                document.tags.map(Tag::name), document.processInstanceId)
    }

    fun mapToGuiObject(document: DBDocument): Document {
        return Document(document.filename,
                document.name,
                document.sender,
                document.owner,
                document.createDate,
                document.tags
                        .map { tag: String? -> Tag(tag!!, document.owner) },
                document.processInstanceId
        )
    }
}